### ISSUE TITLE
`useImportType` should accept `import { type X } from "Y"`

### ISSUE THREAD

**Issue by jasikpark** [biomejs/biome#2005]
### Environment information

```bash
❯ pnpm biome rage
CLI:
  Version:                      1.6.0
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.11.1"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "pnpm/8.14.2"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           true
  Linter disabled:              false
  Organize imports disabled:    true
  VCS disabled:                 false

Workspace:
  Open Documents:               0

```

### Rule name

https://biomejs.dev/linter/rules/use-import-type/

### Playground link

https://biomejs.dev/playground/?code=aQBtAHAAbwByAHQAIAB7ACAAdAB5AHAAZQAgAFgAIAB9ACAAZgByAG8AbQAgACIAWQAiAA%3D%3D

### Expected result

`useImportType` should accept both `import { type X } from "Y"` and `import type { X } from "Y"` since both enable the bundling optimizations.

this is how https://www.typescriptlang.org/tsconfig#verbatimModuleSyntax works in TS

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by Conaclos** [comment]
This is not a bug, this is the intended behavior. `useImportType` factorizes an import that only include inline type imports into `import type`.

**Comment by ematipico** [comment]
Is the documentation explicit enough?

**Comment by Conaclos** [comment]
> Is the documentation explicit enough?

We have this in thje rule description:

> It also groups inline type imports into a grouped import type.

And we have this example:

```ts
import { type A, type B } from "./mod.js";
```

I am open to suggestions, if this can be improved.

**Comment by ematipico** [comment]
Closing. It seems the documentation is explicit enough IMHO

**Comment by torjue** [comment]
Could it be an option to support something like `inline-type-imports`? Ref the eslint rule this is inspired by: [`@typescript-eslint/consistent-type-imports`](https://typescript-eslint.io/rules/consistent-type-imports)

**Comment by throrin19** [comment]
I think same as @torjue . For us is more readable to separate types to normal named imports : 

- Error : 
    ```ts
    import { type A, B } From './a.ts'; 

    const a: A = {};
    const b = new B()
    ```

- Correct : 
   ```ts
   import type { A } from './a.js';
   import { B } From './a.js'; 

    const a: A = {};
    const b = new B()
   ```

**Comment by MarlonPassos-git** [comment]
I use biome in a legacy project that runs TypeScript 3.9.x and there it is not possible to have this type of type import together with normal import. So it would be really cool to have an option to define a preference
